src/EnergyPlus/ScheduleManager.cc

```diff
@@ -506,12 +515,151 @@ namespace ScheduleManager {
 		//because need a week for each day
 		AddDaySch += NumExternalInterfaceFunctionalMockupUnitExportSchedules; // one day schedule for ExternalInterface
 		// to update during run time
+		
+		CurrentModuleObject = "Schedule:File:Shading";
+		NumCommaFileShading = GetNumObjectsFound( CurrentModuleObject );
+		NumAlphas = 0;
+		NumNumbers = 0;
+		if ( NumCommaFileShading > 1 ) {
+			ShowWarningError( CurrentModuleObject + ": More than 1 occurence of this object found, only first will be used." );
+		}
+
+		NumCSVAllColumnsSchedules = 0;
+
+		if ( NumCommaFileShading != 0 ) {
+			GetObjectItem( CurrentModuleObject, 1, Alphas, NumAlphas, Numbers, NumNumbers, Status, lNumericBlanks, lAlphaBlanks, cAlphaFields, cNumericFields );
+			std::string ShadingSunlitFracFileName = Alphas( 1 );
+			CheckForActualFileName( ShadingSunlitFracFileName, FileExists, TempFullFileName );
+			if ( ! FileExists ) {
+				ShowSevereError( RoutineName + ":\"" + ShadingSunlitFracFileName + "\" not found when External Shading Calculation Method = ImportedShading." );
+				ShowContinueError( "Certain run environments require a full path to be included with the file name in the input field." );
+				ShowContinueError( "Try again with putting full path and file name in the field." );
+				ErrorsFound = true;
+			} else {
+				SchdFile = GetNewUnitNumber();
```

Comment:
I know this is how it's done in other places of the code, but I wish I would've seen this earlier and directed to not use this approach.  This file reading approach is simply a carryover from the Fortran days, and using C++ file streams would've been a nice thing to do instead.  But it's a bit late now and not worth holding this up.

Reply:
Okay. I wish I could see this earlier. I thought I should use familiar code to keep the same pattern, as we do everything else.

Reply:
@XuanLuoLBL @Myoldmopar See #6482 followup issue to revise this.